### ISSUE TITLE
[Bug]: Thumbnail class is final, implements no interface

### ISSUE THREAD

**Issue by ychanan** [pimcore/pimcore#13595]
### Expected behavior

`Pimcore\Model\Asset\Image\Thumbnail` class should either be non-final, or it should implement an interface.

### Actual behavior

In our project, we're implementing proxies for various classes, one of which would be the `Pimcore\Model\Asset\Image\Thumbnail` class. The issue we have is that we cannot extend it to build a proxy around it since it is final, and we can't implement a new class since the Thumbnail class does not have an interface.

Additionally, the ImageThumbnailExtension typehits it directly, which pretty much means that there is no way we can use a different class:
`public function getImageThumbnail(Image $image, string $thumbnail, bool $deferred = true): Image\Thumbnail`

### Steps to reproduce

See: https://github.com/pimcore/pimcore/blob/11.x/models/Asset/Image/Thumbnail.php#L28

**Comment by dkarlovi** [comment]
Some theory about the problem: https://ocramius.github.io/blog/when-to-declare-classes-final/

> TL;DR: Make your classes always final, if they implement an interface, and no other public methods are defined

In short, classes should be final if they implement an interface which is then used by the rest of the codebase for types, allowing the class to be replaced / decorated.

**Comment by kjkooistra-youwe** [comment]
The same applies to the ClassDefinition class. Ran into this when trying to mock it for a unit test.

**Comment by dvesh3** [comment]
Fixed by #14337
